### PR TITLE
fix(helpers): disallow reloading cookies if reset cookies is called

### PR DESCRIPTION
**Pull request by xXNoceboXx**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -205,6 +205,16 @@ Scenario: I can fill the step2
     Then I should be on "/step3"
 ```
 
+You can add the tag `behat_helpers_reset_cache` to clear cookies being saved:
```
Comment (mremi) [inline README.md]: `to clear cookies previously saved`

File: README.md [hunk 2]
```diff
@@ -205,6 +205,16 @@ Scenario: I can fill the step2
...
+
+```gherkin
+@behat_helpers_reset_cache
+Scenario: I can't fill the step2
+    Given I fill the second step
```
Comment (mremi) [inline README.md:213]: `second` -> `first`?

**Review by mremi — COMMENTED** [review]
Can you update `CHANGELOG.md` please?

File: README.md [hunk 3]
```diff
@@ -205,6 +205,16 @@ Scenario: I can fill the step2
     Then I should be on "/step3"
 ```
 
+You can add the tag `behat_helpers_reset_cache` to clear cookies previously saved:
+
+```gherkin
+@behat_helpers_reset_cache
+Scenario: I can't fill the step2
```
Comment (mremi) [inline README.md]: I think it should be:
```gherkin
@behat_helpers_reset_cache
Scenario: I am on /step1 if previous cookies are reset
    Given I fill the second step
    Then I should be on "/step3"
    But I am on "/step1"
```

File: src/ReloadCookiesTrait.php [hunk 4]
```diff
@@ -83,7 +83,7 @@ public function doOnce(callable $callback)
             $this->resetCookies();
         }
 
-        if (!$this->cookiesReloaded && $cacheable) {
+        if (!$this->cookiesReloaded && !$reset && $cacheable) {
```
Comment (mremi) [inline src/ReloadCookiesTrait.php:86]: No tests to be added/updated?

**Comment by mremi** [comment]
@xXNoceboXx please rebase

**Comment by mremi** [comment]
@xXNoceboXx any news please?

**Comment by xXNoceboXx** [comment]
@mremi good for you ?

File: tests/ReloadCookiesTraitTest.php [hunk 5]
```diff
@@ -29,6 +30,60 @@ class ReloadCookiesTraitTest extends TestCase
...
+     *
+     * @param $tags
+     * @param $exceptedCookies
+     * @param $exceptedSteps
+     * @param $cookies
+     * @param $steps
+     * @param $resetCounter
+     * @param $sessionCounter
+     * @param $driverCounter
+     * @throws \ReflectionException
```
Comment (mremi) [inline tests/ReloadCookiesTraitTest.php]: to be fixed

File: tests/ReloadCookiesTraitTest.php [hunk 6]
```diff
@@ -15,6 +15,7 @@
 use Behat\Mink\Driver\Selenium2Driver;
 use Behat\Mink\Mink;
 use Behat\Mink\Session;
+use Ekino\BehatHelpers\ReloadCookiesTrait;
```
Comment (mremi) [inline tests/ReloadCookiesTraitTest.php]: useless?

File: tests/ReloadCookiesTraitTest.php [hunk 7]
```diff
@@ -29,6 +30,60 @@ class ReloadCookiesTraitTest extends TestCase
 {
     use TestHelperTrait;
 
+    /**
+     * Test the doOnce method.
+     *
+     * @param $tags
```
Comment (mremi) [inline tests/ReloadCookiesTraitTest.php]: missing type

File: tests/ReloadCookiesTraitTest.php [hunk 8]
```diff
@@ -29,6 +29,59 @@ class ReloadCookiesTraitTest extends TestCase
 {
     use TestHelperTrait;
 
+    /**
+     * Test the doOnce method.
+     *
+     * @param array tags
+     * @param array $exceptedCookies
```
Comment (mremi) [inline tests/ReloadCookiesTraitTest.php]: `$expectedCookies`? and so for the next param

**Comment by xXNoceboXx** [comment]
ping @qsomazzi

File: tests/ReloadCookiesTraitTest.php (hunk 7 above)
Comment (mremi) [reply tests/ReloadCookiesTraitTest.php]: still relevant

File: tests/ReloadCookiesTraitTest.php (hunk 5 above)
Comment (mremi) [reply tests/ReloadCookiesTraitTest.php]: still relevant

File: tests/ReloadCookiesTraitTest.php (hunk 6 above)
Comment (mremi) [reply tests/ReloadCookiesTraitTest.php]: still relevant

**Comment by xXNoceboXx** [comment]
@mremi http://security.sensiolabs.org/check_lock is down

**Comment by mazsudo** [comment]
@xXNoceboXx can you remove end-point in Makefile target here https://github.com/ekino/behat-helpers/blob/374365e0d64a0758fcccaced9bbb4abf158b2deb/Makefile#L21 since it's not necessary. It should be ok after that

File: tests/ReloadCookiesTraitTest.php (hunk 8 above)
Comment (mremi) [reply tests/ReloadCookiesTraitTest.php]: up

**Pull request merged**